spec/system/valuation/budgets_spec.rb

```diff
@@ -1,29 +1,75 @@
 require "rails_helper"
 
 describe "Valuation budgets" do
-  before do
-    valuator = create(:valuator, user: create(:user, username: "Rachel", email: "rachel@valuators.org"))
-    login_as(valuator.user)
-  end
+  let(:valuator) { create(:valuator, user: create(:user)) }
+
+  before { login_as(valuator.user) }
 
   context "Index" do
-    scenario "Displaying budgets" do
-      budget = create(:budget)
+    scenario "Displays published budgets" do
+      create(:budget, name: "Sports")
+      create(:budget, name: "Draft", published: false)
+
+      visit valuation_budgets_path
+
+      expect(page).to have_content("Sports")
+      expect(page).not_to have_content("Draft")
+    end
+
+    scenario "Displays visible and assigned investments count when budget is in valuating phase" do
+      budget = create(:budget, :valuating, name: "Sports")
+      create(:budget_investment, :visible_to_valuators, budget: budget, valuators: [valuator])
+      create(:budget_investment, :invisible_to_valuators, budget: budget, valuators: [valuator])
+      create(:budget_investment, :visible_to_valuators, budget: budget)
+
+      visit valuation_budgets_path
+
+      within "#budget_#{budget.id}" do
+        expect(page).to have_selector(".investments-count", text: "1")
+      end
+    end
+
+    scenario "Displays zero as investments count when budget is not in valuating phase" do
+      budget = create(:budget, %i[accepting finished].sample, name: "Sports")
+      create(:budget_investment, :visible_to_valuators, budget: budget, valuators: [valuator])
```

Comment:
Since these tests (and the next one) don't test user interaction or controller filters, what do you think about moving them to component tests? :thinking:

Makes sense to keep the "Displays published budgets" test here because it tests controller filters and thanks to it we've got one system test accessing the valuation index.

Reply:
Much better! Thanks